test/SmokeTests/OOProcSmokeTests/DotNetIsolated/run-smoke-tests.ps1

```diff
@@ -0,0 +1,115 @@
+)
+
+$retryCount = 0;
+$statusUrl = $null;
+$success = $false;
+$haveManuallyRestartedHost = $false;
+
+Do {
+    $testIsRunning = $true;
+
+    # Start the functions host if it's not running already.
+    # Then give it up to 1 minute to start up. 
+    # This is a long wait, but from experience the CI can be slow to start up the host, especially after a platform-error.
+    $isFunctionsHostRunning = (Get-Process -Name func -ErrorAction SilentlyContinue)
+    if ($isFunctionsHostRunning -eq $null) {
+        Write-Host "Starting the Functions host..." -ForegroundColor Yellow
+
+        # The '&' operator is used to run the command in the background
+        cd ./test/SmokeTests/OOProcSmokeTests/DotNetIsolated && func host start --port 7071 &       
+        Write-Host "Waiting for the Functions host to start up..." -ForegroundColor Yellow
+        Start-Sleep -Seconds 60
+    }
+
+    
+    try {
+        # Make sure the Functions runtime is up and running
+        $pingUrl = "http://localhost:7071/admin/host/ping"
+        Write-Host "Pinging app at $pingUrl to ensure the host is healthy" -ForegroundColor Yellow
+        Invoke-RestMethod -Method Post -Uri "http://localhost:7071/admin/host/ping"
+        Write-Host "Host is healthy!" -ForegroundColor Green
+
+        # Start orchestrator if it hasn't been started yet
+        if ($statusUrl -eq $null){
+            $startOrchestrationUri = "http://localhost:7071/$HttpStartPath"
+            Write-Host "Starting a new orchestration instance via POST to $startOrchestrationUri..." -ForegroundColor Yellow
+
+            $result = Invoke-RestMethod -Method Post -Uri $startOrchestrationUri
+            Write-Host "Started orchestration with instance ID '$($result.id)'!" -ForegroundColor Yellow
+            Write-Host "Waiting for orchestration to complete..." -ForegroundColor Yellow
+
+            $statusUrl = $result.statusQueryGetUri
+            
+            # sleep for a bit to give the orchestrator a chance to start,
+            # then loop once more in case the orchestrator ran quickly, made the host unhealthy,
+            # and the functions host needs to be restarted
+            Start-Sleep -Seconds 5
+            continue;
+        }
+
+        # Check the orchestrator status
+        $result = Invoke-RestMethod -Method Get -Uri $statusUrl
+        $runtimeStatus = $result.runtimeStatus
+        Write-Host "Orchestration is $runtimeStatus" -ForegroundColor Yellow
+        Write-Host $result
+
+        if ($result.runtimeStatus -eq "Completed") {
+            $success = $true
+            $testIsRunning = $false
+            break
+        }
+        if ($result.runtimeStatus -eq "Failed") {
+            $success = $false
+            $testIsRunning = $false
+            break
+        }
+
+        # If the orchestrator did not complete yet, wait for a bit before checking again
+        Start-Sleep -Seconds 2
+        $retryCount = $retryCount + 1
+
+    } catch {
```

Comment:
When is it expected that we would enter this `catch` block?

Reply:
I added comments to the script clarifying this here: https://github.com/Azure/azure-functions-durable-extension/pull/2922/commits/2ff4b79d571e2ec288e5e2da17db92c8370b23a1

Long story short - we expect to enter this catch block if any HTTP request to the host fails. This can happen if the host has not started yet, or if has become unresponsive for whatever reason. Recall that we're inducing isolated worker process crashes, and in my experience that can make the host unresponsive for a bit.

Reply:
Got it, I didn't realize that `Invoke-RestMethod` raised an error on non-success status codes.